arangod/Statistics/StatisticsFeature.cpp

```diff
@@ -186,19 +194,27 @@ void StatisticsFeature::start() {
   }
 
   _statisticsThread.reset(new StatisticsThread);
-  _statisticsWorker.reset(new StatisticsWorker(*vocbase));
 
   if (!_statisticsThread->start()) {
     LOG_TOPIC(FATAL, arangodb::Logger::STATISTICS)
         << "could not start statistics thread";
     FATAL_ERROR_EXIT();
   }
 
-  if (!_statisticsWorker->start()) {
-    LOG_TOPIC(FATAL, arangodb::Logger::STATISTICS)
+  // force history disable on Agents
+  if (arangodb::ServerState::instance()->isAgent()) {
```

Comment:
Is the role always known at this point?

Reply:
Can we make sure here programmatically that the server role has already been determined, and that this also works in the cluster?

Reply:
done